homeassistant/components/reolink/entity.py

```diff
@@ -89,11 +89,21 @@ def __init__(self, reolink_data: ReolinkData) -> None:
     async def async_added_to_hass(self) -> None:
         """Entity created."""
         await super().async_added_to_hass()
-        if (
-            self.entity_description.cmd_key is not None
-            and self.entity_description.cmd_key not in self._host.update_cmd_list
-        ):
-            self._host.update_cmd_list.append(self.entity_description.cmd_key)
+        cmd_key = self.entity_description.cmd_key
+        if cmd_key is not None:
+            self._host.update_cmd_list_count[cmd_key].setdefault("host", 0)
+            self._host.update_cmd_list_count[cmd_key]["host"] += 1
+            self._host.update_cmd_list.setdefault(cmd_key, [])
```

Comment:
Do you need a seperate `list` and `count`. Can you use the same container where the value is the reference count, and the key is the cmd_key?

Reply:
well I also need the channel in there.
The upstream library needs to now which commands it needs to send to which channels.
(the library talks to a NVR which has multiple camera's connected using a build-in POE switch, so each channel is a seperate camera)

Currently the structure for the upstream library is:
{cmd_key: [channels]}
So for instance if only camera's on channel 2 and 5 support Zoom and those entities are enabled:
{"GetZoomFocus": [2, 5]}
For commands that talk to the NVR itself and not to a channel, for instance the hard drive status of the NVR, it looks like:
{"GetHddInfo": []}

Reply:
The update_cmd_list_count looks like:
{"GetZoomFocus": {"Host": 6, 2: 3, 5: 3}}
and
{"GetHddInfo": {"Host": 1}}

But that added complexity of knowing how many entities are using the command is not needed in the upstream library (reolink-aio)

Reply:
https://github.com/bdraco/home-assistant/commit/272e28efbbf2e2d3043e638fc12ed87b709a9080

Maybe something like this